airflow/models/xcom.py

```diff
@@ -154,6 +154,9 @@ def set(
         """
         from airflow.models.dagrun import DagRun
 
+        if not key:
+            raise ValueError(f"XCom key must be a non-empty string. Received: {key}")
```

Comment:
```suggestion
            raise ValueError(f"XCom key must be a non-empty string. Received: {key!r}")
```

Otherwise it will be woefully unclear what value is passed